### PR TITLE
Added support for a retry interval to monitors

### PR DESCRIPTION
**Pull request by No0Vad**
If a check fails and retries are used you can now specify a specific value for that. So you can check faster if the site goes back up again.

Also added some more text for the log.

This should also complete the requested feature here #305

### Images
image
image

### Sample output to the log
```
Monitor #1 'Test': Successful Response: 17 ms | Interval: 60 seconds | Type: http
Monitor #1 'Test': Successful Response: 21 ms | Interval: 60 seconds | Type: http
Monitor #1 'Test': Successful Response: 25 ms | Interval: 60 seconds | Type: http
Monitor #1 'Test': Successful Response: 21 ms | Interval: 60 seconds | Type: http
Monitor #1 'Test': Pending: Request failed with status code 404 | Max retries: 5 | Retry: 1 | Retry Interval: 5 seconds | Type: http
Monitor #1 'Test': Pending: Request failed with status code 404 | Max retries: 5 | Retry: 2 | Retry Interval: 5 seconds | Type: http
Monitor #1 'Test': Pending: Request failed with status code 404 | Max retries: 5 | Retry: 3 | Retry Interval: 5 seconds | Type: http
Monitor #1 'Test': Pending: Request failed with status code 404 | Max retries: 5 | Retry: 4 | Retry Interval: 5 seconds | Type: http
Monitor #1 'Test': Successful Response: 741 ms | Interval: 60 seconds | Type: http
Monitor #1 'Test': Successful Response: 23 ms | Interval: 60 seconds | Type: http
Monitor #1 'Test': Pending: Request failed with status code 404 | Max retries: 5 | Retry: 1 | Retry Interval: 5 seconds | Type: http
Monitor #1 'Test': Pending: Request failed with status code 404 | Max retries: 5 | Retry: 2 | Retry Interval: 5 seconds | Type: http
Monitor #1 'Test': Pending: Request failed with status code 404 | Max retries: 5 | Retry: 3 | Retry Interval: 5 seconds | Type: http
Monitor #1 'Test': Pending: Request failed with status code 404 | Max retries: 5 | Retry: 4 | Retry Interval: 5 seconds | Type: http
Monitor #1 'Test': Pending: Request failed with status code 404 | Max retries: 5 | Retry: 5 | Retry Interval: 5 seconds | Type: http
Monitor #1 'Test': Failing: Request failed with status code 404 | Interval: 60 seconds | Type: http
Monitor #1 'Test': Successful Response: 730 ms | Interval: 60 seconds | Type: http
Monitor #1 'Test': Successful Response: 23 ms | Interval: 60 seconds | Type: http
```

I'm really enjoy what you have done with Uptime Kuma, well done! 👏
I wanted to contribute, so I had to figure out how to do pull requests. Hopefully this works 🙏

### CODE REVIEW

File: server/database.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ class Database {
     static patchList = {
         "patch-setting-value-type.sql": true,
         "patch-improve-performance.sql": true,
+        "patch-add-retry-interval-monitor.sql": true
```
Comment (Saibamen) [inline server/database.js:33]:
```suggestion
        "patch-add-retry-interval-monitor.sql": true,
```

**Review by Saibamen — CHANGES_REQUESTED** [review]

File: server/database.js (hunk 1 above)
Comment (No0Vad) [reply server/database.js:33]: I think I'm missing something here? But a `,` is not needed on the last key/value pair of a object.

Comment (Saibamen) [reply server/database.js:33]: We have `comma-dangle` enabled in ESLint (https://eslint.org/docs/rules/comma-dangle)

Comment (No0Vad) [reply server/database.js:33]: Ahh, I see now. My VSCode did not tell me that was a problem (ESLint extension is installed). Hmm strange 🤔

Do I make the changes locally, or press the "Commit suggestions" to use your suggested changes?
(New to contributing on projects on GitHub 😅)

Comment (Saibamen) [reply server/database.js:33]: Use commit suggestions button. This is faster than editing file, commiting and pushing ;)

Comment (No0Vad) [reply server/database.js]: Thank you for the help ☺

**Comment by louislam** [comment]
Nice feature.

However, it may causes unexpected behaviors if the interval is too low (e.g. 1 second). It changed a bit:
- The minimum value of Retry interval = 20s
- Link `Interval` and `RetryInterval` if they are the same.

**Comment by No0Vad** [comment]
Thanks!

Sounds like a good suggestion, I will check it out after work today.
I might have to tweak the texts a bit too.

**Comment by chakflying** [comment]
@No0Vad I just tried to update, but found that it's showing 0 seconds which seems quite confusing. Is this the intended behavior?

image

**Comment by louislam** [comment]
> @No0Vad I just tried to update, but found that it's showing 0 seconds which seems quite confusing. Is this the intended behavior?
> 
> image

Thanks, I forgot to check the existing monitor.

**Pull request merged**